From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 13 Sep 2024 13:49:03 +0200
Subject: dbus: Add Backlight property to DisplayConfig
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

This ensures GNOME 47's gsd-power can read it and doesn't crash.

Having it empty is fine for the moment.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1526>
---
 src/dbus/org.gnome.Mutter.DisplayConfig.xml | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/dbus/org.gnome.Mutter.DisplayConfig.xml b/src/dbus/org.gnome.Mutter.DisplayConfig.xml
index 0501908..fe0d106 100644
--- a/src/dbus/org.gnome.Mutter.DisplayConfig.xml
+++ b/src/dbus/org.gnome.Mutter.DisplayConfig.xml
@@ -216,6 +216,22 @@
       <arg name="new_value" direction="out" type="i" />
     </method>
 
+    <!--
+	Backlight:
+
+	A set of backlights. Each backlight is a dictionary with the following
+	entries. If an entry is only a connector, it means there is no way to
+	control it via this D-Bus interface.
+
+	  * 'connector' (s) - An associated monitor connector
+	  * 'active' (s) - True if the monitor is active
+	  * 'min_step' (i) - Minimum step amount (optional)
+	  * 'value' (i) - Current value (optional)
+
+	The initial 'u' is a serial number used when setting the backlight.
+    -->
+    <property name="Backlight" type="(uaa{sv})" access="read" />
+
     <!--
         GetCrtcGamma:
 	@serial: configuration serial
